analysis/analysis.go

```diff
@@ -127,6 +127,20 @@ func generateMetricTasks(run *v1alpha1.AnalysisRun) []metricTask {
 			continue
 		}
 		if lastMeasurement == nil {
+			if metric.StartDelay != "" {
+				if run.Status.StartedAt == nil {
+					continue
+				}
+				duration, err := metric.StartDelay.Duration()
+				if err != nil {
+					logCtx.Warnf("failed to parse duration: %v", err)
+					continue
```

Comment:
This is fine here, but can we also have this fail validation of the rollout itself (before reconciliation)?

Reply:
I already have it a part of validation here: https://github.com/argoproj/argo-rollouts/pull/350/files?file-filters%5B%5D=.go#diff-11d6da907459856384573d1855393e33R99-R103. Is this what you mean?

Reply:
You're right. Ignore this.